.github/workflows/build-and-deploy.yml

```diff
@@ -3,7 +3,7 @@ name: Deploy Coinbase Wallet SDK Playground to Pages
 on:
```

Comment:
hello reviewers. 
here are some contexts of this PR that i hope you find useful for your review.

1. `packages/playground` folder has the files from [playground branch](https://github.com/coinbase/coinbase-wallet-sdk/tree/playground)
other than that, just config changes.
![image](https://github.com/user-attachments/assets/654098e2-b3b4-481b-80be-32b038dead71)

2. there are [some commits cherrypicked](https://github.com/coinbase/coinbase-wallet-sdk/pull/1398/commits) to sync with playground branch. 
and some changes in the CI script to auto deploy playground on each commit 
and minor tweak to provide `HEAD` as an SDK version option in the playground
![image](https://github.com/user-attachments/assets/1d789aef-976a-4dfe-85db-7095d68a3922)